lib/telephony/pinpoint/opt_out_manager.rb

```diff
@@ -0,0 +1,48 @@
+module Telephony
+  module Pinpoint
+    class OptOutManager
+      # @return [Response]
+      def opt_in_phone_number(phone_number)
+        if Telephony.config.pinpoint.sms_configs.empty?
+          return PinpointHelper.handle_config_failure(:sns)
+        end
+
+        response = nil
+        Telephony.config.pinpoint.sms_configs.each do |config|
+          client = build_client(config)
+          next if client.nil?
+
+          opt_in_response = client.opt_in_phone_number(phone_number: phone_number)
+
+          return Response.new(success: opt_in_response.successful?)
+        rescue Aws::SNS::Errors::InvalidParameter
+          # This is thrown when the number has been opted in too recently
+          return Response.new(success: false)
+        rescue Seahorse::Client::NetworkingError,
+               Aws::SNS::Errors::ServiceError => error
+          PinpointHelper.notify_pinpoint_failover(
+            error: error,
+            region: config.region,
+            channel: :sns,
```

Comment:
Just to confirm: Is this supposed to be `:sns` or `:sms`? Same question on line 7 above.

Reply:
yup it's `sns` because this uses SNS client, this is only for our own monitoring, so I could call it `notification_service` if that would be clearer?

Reply:
> yup it's `sns` because this uses SNS client, this is only for our own monitoring, so I could call it `notification_service` if that would be clearer?

Cool, just wanted to double-check, since I only ever seen `channel: :sms` and `channel: :voice` elsewhere. Up to you about using `notification_service` instead, no preference from me.

Reply:
expanded in b1349324d421504b3977f0b6264c58c69f98a875